File: solutions/2285.cpp
class Solution {
public:
    long long maximumImportance(int n, vector<vector<int>>& roads) {
        vector<int> outDegree(n, 0);
        // iterate each edge and collect outDegree
        for (auto road : roads) {
            outDegree[road[0]]++;
            outDegree[road[1]]++;
        }
        sort(outDegree.begin(), outDegree.end());
        // assign low node number to those node with low outDegree
        long long sum = 0;
        for (int i = 0; i < n; ++i) {
            sum += outDegree[i] * (long long)(i + 1);
        }
        return sum;
    }
};

File: rust_solutions/2285.rs
impl Solution {
    pub fn maximum_importance(n: i32, roads: Vec<Vec<i32>>) -> i64 {
        let mut indegrees = vec![0; n as usize];
        let m = roads.len();
        for i in 0..m {
            indegrees[roads[i][0] as usize] += 1;
            indegrees[roads[i][1] as usize] += 1;
        }
        indegrees.sort();
        let mut res: i64 = 0;
        for i in 0..n {
            res += (indegrees[i as usize] as i64) * ((i + 1) as i64);
        }
        res
    }
}

